diff --git a/library/think/cache/driver/Redis.php b/library/think/cache/driver/Redis.php
index 8956e774..7fb60b57 100644
--- a/library/think/cache/driver/Redis.php
+++ b/library/think/cache/driver/Redis.php
@@ -162,7 +162,7 @@ class Redis extends Driver
      */
     public function rm($name)
     {
-        return $this->handler->delete($this->getCacheKey($name));
+        return $this->handler->del($this->getCacheKey($name));
     }
 
     /**
